modules/govuk_mysql/manifests/xtrabackup/backup.pp

```diff
@@ -0,0 +1,113 @@
+# == Class: Govuk_mysql::Xtrabackup::Backup
+#
+# Creates base and incremental backups which are streamed to an S3 bucket.
+# Based upon this MariaDB blog post: https://mariadb.com/blog/streaming-mariadb-backups-cloud
+#
+# === Parameters
+#
+# [*aws_access_key_id*]
+#   The AWS access key used to access the S3 bucket.
+#
+# [*aws_secret_access_key*]
+#   The AWS secret access key used to access the S3 bucket.
+#
+# [*aws_region*]
+#   The region where the S3 bucket lives.
+#   Default: eu-west-1
+#
+# [*s3_bucket_name*]
+#   The unique name of the S3 bucket.
+#
+# [*encryption_key*]
+#   The encryption key used to encrypt the backups.
+#
+# [*base_backup_cron_minute*]
+#   The minute(s) that the daily base backup(s) run(s).
+#   Default: 10
+#
+# [*base_backup_cron_hour*]
+#   The hour(s) that the daily base backup(s) run(s).
+#   Default: 6
+#
+# [*incremental_backup_cron_minute*]
+#   The minute(s) that the incremental backup(s) run(s).
+#   Default: */15 (every 15 minutes)
+#
+# [*enabled*]
+#   If this is true then the packages are installed and backups are triggered.
+#   Default: false
+#
+class govuk_mysql::xtrabackup::backup (
+  $aws_access_key_id = undef,
+  $aws_secret_access_key = undef,
+  $aws_region = 'eu-west-1',
+  $s3_bucket_name = undef,
+  $encryption_key = undef,
+  $base_backup_cron_minute = '10',
+  $base_backup_cron_hour = '6',
+  $incremental_backup_cron_minute = '*/15',
+  $enabled = false,
+) {
+  if $enabled {
+    include govuk_mysql::xtrabackup::packages
+
+    file {['/etc/mysql/xtrabackup', '/etc/mysql/xtrabackup/env.d']:
+      ensure => 'directory',
+      mode   => '0755',
+      owner  => 'root',
```

Comment:
We set the `owner` and `group` to `root` [by default](https://github.com/alphagov/govuk-puppet/blob/d41593b133bfc1a2984cb8b4870f2fefd79e7377/manifests/site.pp#L3-L7).